mkosi/__init__.py

```diff
@@ -1387,6 +1401,8 @@ def mount_image(
                     os.rename(realroot / entry.name, root / entry.name)
                 realroot.rmdir()
 
+                delete_whiteout_files(root)
```

Comment:
I think it's reasonable to do this here. Those files are very specific to overlayfs, and since we're cleaning up after overlayfs here, it seems like the appropriate place to remove those too.